src/Listener/RelatedModelsListener.php

```diff
@@ -87,7 +87,7 @@ public function publishRelatedModels($action = null, $entity = null)
         foreach ($models as $name => $association) {
             list(, $associationName) = pluginSplit($association->name());
             $viewVar = Inflector::variable($associationName);
-            if (array_key_exists($viewVar, $controller->viewVars)) {
+            if (array_key_exists($viewVar, (array)$controller->viewVars)) {
```

Comment:
> It is not bound, but if someone does the effort of doing it right it should not be "shut down" just because, but welcomed IMO :/

You're assuming the Core is 'doing it right', you're assuming that you are being 'shut down', and you're assuming your contribution is not welcome. 😉